### ISSUE TITLE
UnitTestRunner fails with CONFIG=Release on Linux/amd64/gcc

### ISSUE THREAD

**Issue by umlaeute** [WeAreROLI/JUCE#380]
I'm compiling JUCE-5.3.1 myself, and for testing purposes I also build and run the `UnitTestRunner`.

The system is Debian/sid on amd64, using gcc:

```
$ uname -a
Linux xenakis 4.15.0-2-amd64 #1 SMP Debian 4.15.11-1 (2018-03-20) x86_64 GNU/Linux
$ g++ --version
g++ (Debian 7.3.0-16) 7.3.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

Now compiling the UnitTestRunner with `CONFIG=Debug` basically makes the unit-tests succeed (there seems to be a small glitch with 1 of the 22 Files/Reading tests. for now i'll ignore that, as i think this is temporary).
However, when compiling with `CONFIG=Release`, many of t he `SIMDRegister` tests fail.
here's an excerpt of the output:

```
Starting test: SIMDRegister UnitTests / InitializationTest...
FAILED!!  5 tests failed, out of a total of 24
-----------------------------------------------------------------
Starting test: SIMDRegister UnitTests / AccessTest...
FAILED!!  5 tests failed, out of a total of 81
-----------------------------------------------------------------
Starting test: SIMDRegister UnitTests / AdditionOperators...
FAILED!!  4992 tests failed, out of a total of 12000
-----------------------------------------------------------------
Starting test: SIMDRegister UnitTests / SubtractionOperators...
FAILED!!  5215 tests failed, out of a total of 12000
-----------------------------------------------------------------
Starting test: SIMDRegister UnitTests / MultiplicationOperators...
FAILED!!  4980 tests failed, out of a total of 12000
-----------------------------------------------------------------
Starting test: SIMDRegister UnitTests / BitANDOperators...
FAILED!!  5684 tests failed, out of a total of 13200
-----------------------------------------------------------------
Starting test: SIMDRegister UnitTests / BitOROperators...
FAILED!!  6196 tests failed, out of a total of 13200
-----------------------------------------------------------------
Starting test: SIMDRegister UnitTests / BitXOROperators...
FAILED!!  6195 tests failed, out of a total of 13200
-----------------------------------------------------------------
Starting test: SIMDRegister UnitTests / CheckComparisons...
FAILED!!  3946 tests failed, out of a total of 18000
-----------------------------------------------------------------
Starting test: SIMDRegister UnitTests / CheckMinMax...
FAILED!!  2000 tests failed, out of a total of 4000
-----------------------------------------------------------------
Starting test: SIMDRegister UnitTests / CheckMultiplyAdd...
FAILED!!  5 tests failed, out of a total of 12
-----------------------------------------------------------------
Starting test: SIMDRegister UnitTests / CheckSum...
All tests completed successfully
```

also the UnitTest likes to segfault randomly (the output above is obviously from a run without a segfault).

compiling the UnitTest (CONFIG=Release) with `clang` behaves normally (no issues with the SIMDRegister test).

```
$ clang++ --version
clang version 4.0.1-10 (tags/RELEASE_401/final)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```

**Comment by hogliux** [comment]
Thanks for the bug report. We'll sort this out...

**Comment by hogliux** [comment]
OK this is now fixed with commit `[b719acf](https://github.com/WeAreROLI/JUCE/commit/b719acf3e228a2821e4f4934e4e3b1710eafb51d)`. Thank you for reporting!

**Comment by hogliux** [comment]
OK this is now fixed with commit `[b719acf](https://github.com/WeAreROLI/JUCE/commit/b719acf3e228a2821e4f4934e4e3b1710eafb51d)`. Thank you for reporting!
